jerry-core/parser/js/byte-code.h

```diff
@@ -821,16 +821,59 @@ typedef enum
   CBC_CODE_FLAGS_MAPPED_ARGUMENTS_NEEDED = (1u << 4), /**< mapped arguments object must be constructed */
   CBC_CODE_FLAGS_UNMAPPED_ARGUMENTS_NEEDED = (1u << 5), /**< mapped arguments object must be constructed */
   CBC_CODE_FLAGS_LEXICAL_ENV_NOT_NEEDED = (1u << 6), /**< no need to create a lexical environment */
-  CBC_CODE_FLAGS_ARROW_FUNCTION = (1u << 7), /**< this function is an arrow function */
-  CBC_CODE_FLAGS_STATIC_FUNCTION = (1u << 8), /**< this function is a static snapshot function */
-  CBC_CODE_FLAGS_DEBUGGER_IGNORE = (1u << 9), /**< this function should be ignored by debugger */
-  CBC_CODE_FLAGS_CLASS_CONSTRUCTOR = (1u << 10), /**< this function is a class constructor */
-  CBC_CODE_FLAGS_GENERATOR = (1u << 11), /**< this function is a generator */
-  CBC_CODE_FLAGS_REST_PARAMETER = (1u << 12), /**< this function has rest parameter */
-  CBC_CODE_FLAG_HAS_TAGGED_LITERALS = (1u << 13), /**< this function has tagged template literal list */
-  CBC_CODE_FLAGS_LEXICAL_BLOCK_NEEDED = (1u << 14), /**< compiled code needs a lexical block */
-  CBC_CODE_FLAGS_ACCESSOR = (1u << 15) /**< accessor propety 'get' and 'set' functions */
-} cbc_code_flags;
+  CBC_CODE_FLAGS_STATIC_FUNCTION = (1u << 7), /**< this function is a static snapshot function */
+  CBC_CODE_FLAGS_DEBUGGER_IGNORE = (1u << 8), /**< this function should be ignored by debugger */
+  CBC_CODE_FLAGS_REST_PARAMETER = (1u << 9), /**< this function has rest parameter */
+  CBC_CODE_FLAGS_HAS_TAGGED_LITERALS = (1u << 10), /**< this function has tagged template literal list */
+  CBC_CODE_FLAGS_LEXICAL_BLOCK_NEEDED = (1u << 11), /**< compiled code needs a lexical block */
+
+  /* Bits from bit 13 is reserved for function types (see CBC_CODE_FLAGS_FUNCTION_TYPE_SHIFT). */
+} cbc_code_flags_t;
+
+/**
+ * Compact byte code function types.
+ */
+typedef enum
+{
+  CBC_FUNCTION_NORMAL, /**< function without special properties */
```

Comment:
Cannot we extend this list with Regexp literals? `CBC_CODE_FLAGS_FUNCTION` is only used to represent that the compiled code is not RegExp, so IMO this should be a new type.